### PR TITLE
Modify regex to correctly split compilation error in test files

### PR DESCRIPTION
**Pull request by ninewise**
The old regex used the name of the tested file to split the compilation
messages. Compilation errors in the test files could not be split,
resulting in an internal error.

### CODE REVIEW

File: run [hunk 1]
```diff
@@ -243,7 +243,7 @@ find "$judge/src" -name '*.java' \
 jar -cf "judge.jar" -C /tmp/build .
 
 # Compiling the tests
-if ! find "$resources" -name '*.java' | xargs javac -cp ".:${resources}:${worklibs}:${testlibs}:judge.jar" -d . -sourcepath "$resources" > "$compilation" 2>&1; then
+if ! find "$resources" -name '*.java' | xargs javac -Xdiags:verbose -cp ".:${resources}:${worklibs}:${testlibs}:judge.jar" -d . -sourcepath "$resources" > "$compilation" 2>&1; then
```
Comment (charvp) [inline run]: Is this relevant to the PR? (It doesn't seem like it is.)

Comment (ninewise) [reply run]: It is, but I forgot to mention it in the commit message; I'll modify it and push after fixing some other stuff.
